src/ucp/api/ucp.h

```diff
@@ -2401,6 +2405,46 @@ ucs_status_ptr_t ucp_am_send_nb(ucp_ep_h ep, uint16_t id,
                                 ucp_send_callback_t cb, unsigned flags);
 
 
+/**
+ * @ingroup UCP_COMM
+ * @brief Send Active Message.
+ *
+ * This routine sends an Active Message to an ep. If the operation completes
+ * immediately and UCP_OP_ATTR_FLAG_NO_IMM_CMPL flag is not set in op_attr_mask
+ * field of @a param, the routine returns UCS_OK and the call-back routine is
+ * not invoked even if specified. Otherwise, if no error is reported and
+ * UCP_OP_ATTR_FIELD_CALLBACK flag is passed in op_attr_mask field of @a param,
+ * then the UCP library will schedule invocation of the call-back routine
+ * @a param->cb.send upon completion of the operation.
+ *
+ * @note This operation does not support CUDA memory.
+ * @note This operation supports specific flags, which can be passed
+ *       in @a param. The exact set of flags is defined by @ref ucp_send_am_flags.
+ *
+ * @param [in]  ep          UCP endpoint where the Active Message will be run.
+ * @param [in]  id          Active Message id. Specifies which registered
+ *                          callback to run.
+ * @param [in]  buffer      Pointer to the data to be sent to the target node
+ *                          of the Active Message.
+ * @param [in]  count       Number of elements to send.
+ * @param [in]  param       Operation parameters, see @ref ucp_request_param_t.
+ *
+ *
+ * @return NULL                 - Active Message was sent immediately.
```

Comment:
looks like doesn't match with the description (`... returns UCS_OK`):
```
... If the operation completes immediately and UCP_OP_ATTR_FLAG_NO_IMM_CMPL flag is not set in op_attr_mask field of @a param, the routine returns UCS_OK ...
```
`@return` section:
```
@return NULL                 - Active Message was sent immediately.
```

pls, correct me if I'm wrong

Reply:
yes, it is a bit confusing (even though it means the same). But we have this mismatch in many other communication routines. Let's wait for community review, maybe would need to update all funcs then.

Reply:
@brminich yes, sounds confusing. ok :)

Reply:
@yosefe, wdyt about the this wording? Shall we state that the func returns `UCS_OK` rather than `NULL` if completed immediately ?

Reply:
it's ucs_status_ptr_t , so NULL